ksql-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilder.java

```diff
@@ -314,7 +315,13 @@ private static TimestampExtractor timestampExtractor(
         .map(Column::index)
         .orElse(-1);
 
-    return timestampPolicy.create(timestampIndex);
+    return timestampPolicy.create(
+        timestampIndex,
+        ksqlConfig.getBoolean(KsqlConfig.KSQL_TIMESTAMP_THROW_ON_INVALID),
+        queryBuilder.getProcessingLogContext()
+            .getLoggerFactory()
+            .getLogger(queryBuilder.getQueryId().getId())
```

Comment:
Use this for the logger name instead of the id directly:

```
final QueryId queryId = queryBuilder.getQueryId();
final QueryContext queryContex = source.getProperties().getQueryContext()
final String loggerNamePrefix = QueryLoggerUtil.queryLoggerName(queryId, queryContext);
```

This way you will get a more specific logger prefix that tells you where in the query plan the message was generated.